### ISSUE TITLE
Dotty needs to support overloaded methods that have same erazed arguments.

### ISSUE THREAD

**Issue by DarkDimius** [lampepfl/dotty#808]
Method is inherited from scala2 trait BCodeIdiomatic.
Seems to be a problem with overloads, as there should be 3 mkArray methods:

```
  final def mkArray(xs: List[BType]): Array[BType]
  final def mkArray(xs: List[String]): Array[String]
  final def mkArray(xs: List[asm.Label]): Array[asm.Label]
```

but only 
```
  public final scala.tools.nsc.backend.jvm.BTypes$BType[] mkArray(scala.collection.immutable.List)
```
is actually emitted in bytecode.

**Comment by DarkDimius** [comment]
@odersky: a minimized example 

```scala
// to be compiled by scalac
trait T{
 def foo(a: List[String]): String = "1"
 def foo(a: List[Int]): Int = 1
 foo(List("1"))
 foo(List(1))
}
```

```scala
// to be compiled by dotty
object Test extends T{
  def main(args: Array[String]): Unit = ()
}
```

**Comment by odersky** [comment]
Was fixed by #812
